### ISSUE TITLE
out of control RAM usage

### ISSUE THREAD

**Issue by ThomasMeetKai** [geziyor/geziyor#28]
I've got a script that clicks every link and then clicks every link, and it quickly gets out of hand in terms of memory usage (40+GB) before crashing. Any suggestions as to where it's getting out of control? Storing millions of requests shouldn't take that much RAM in my mind.

**Comment by ThomasMeetKai** [comment]
A memory profile seems to indicate half of it is happening here 
![image](https://user-images.githubusercontent.com/82546848/123468408-56fbb500-d5b7-11eb-82e6-b538a4d6bc97.png)

**Comment by ThomasMeetKai** [comment]
should io.Copy be used instead of ReadAll?

**Comment by musabgultekin** [comment]
There could be some kind of memory leak. I experienced the same issue when I used it on more than 10 million requests. Like It's storing all body data In the heap? I don't know.

Actually another Go scraper uses io.ReadAll https://github.com/gocolly/colly/blob/2f09941613011bfde62cbe4a695310b42bf42d41/http_backend.go#L210

Maybe we just need to implement some kind of queue system. I remember Python Scrapy uses some kind of queue system.
